indexing-service/src/main/java/org/apache/druid/indexing/common/task/CompactionTask.java

```diff
@@ -181,19 +177,10 @@ public CompactionTask(
       throw new IAE("Interval[%s] is empty, must specify a nonempty interval", interval);
     }
 
-    if ((keepSegmentGranularity != null && keepSegmentGranularity) && segmentGranularity != null) {
-      throw new IAE("keepSegmentGranularity and segmentGranularity can't be used together");
-    }
-
-    if (keepSegmentGranularity != null) {
-      log.warn("keepSegmentGranularity is deprecated. Set a proper segmentGranularity instead");
-    }
-
     this.interval = interval;
     this.segments = segments;
     this.dimensionsSpec = dimensionsSpec == null ? dimensions : dimensionsSpec;
     this.metricsSpec = metricsSpec;
-    this.keepSegmentGranularity = keepSegmentGranularity;
     this.segmentGranularity = segmentGranularity;
```

Comment:
Please include a null check for `segmentGranularity` with a nice error message. It's especially important since people with older ingest specs that _don't_ include segmentGranularity need a nice message telling them what to change.

Reply:
`segmentGranularity` is nullable. Added the annotation properly.